prefork/prefork.go

```diff
@@ -110,30 +120,84 @@ func (p *Prefork) setTCPListenerFiles(addr string) error {
 	return nil
 }
 
-func (p *Prefork) prefork(addr string) error {
-	chErr := make(chan error, 1)
-
+func (p *Prefork) prefork(addr string) (err error) {
 	if !p.Reuseport {
-		if err := p.setTCPListenerFiles(addr); err != nil {
-			return err
+		if err = p.setTCPListenerFiles(addr); err != nil {
+			return
 		}
 
-		defer p.ln.Close()
+		defer func() {
+			if err == nil {
+				err = p.ln.Close()
```

Comment:
Can you document why the listener only being closed if `!p.Reuseport`.

Reply:
And even if `err != nil` the listener still needs to be closed properly.

Reply:
> Can you document why the listener only being closed if !p.Reuseport.

You mean to add some comments on it?

Reply:
Yes please. I know the close was already there, but might as well add a comment if we're making changes here 😄

Reply:
Done.